### DISCUSSION TITLE
Welcome to Solana-Memecoin-CLI Discussions!

### DISCUSSION THREAD

**Discussion by outsmartchad** [outsmartchad/outsmart-cli#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.
  * To develop the project together: [https://discord.gg/bB2n9bJU](url)

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by pandaamx** [comment]
Nice project!
Have discord or another way to contact you directly?

**Reply by pandaamx** [reply]
I received an error when using the create.js file.

node:internal/modules/cjs/loader:1189
  throw err;
  ^

Error: Cannot find module '@metaplex-foundation/umi-bundle-defaults'
Require stack:
- C:\Users\Arthur Araujo\Desktop\Solana-Memecoin-CLI\src\Token\create.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1186:15)
    at Module._load (node:internal/modules/cjs/loader:1012:27)
    at Module.require (node:internal/modules/cjs/loader:1271:19)
    at require (node:internal/modules/helpers:123:16)
    at Object.<anonymous> (C:\Users\Arthur Araujo\Desktop\Solana-Memecoin-CLI\src\Token\create.js:19:23)
    at Module._compile (node:internal/modules/cjs/loader:1434:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1518:10)
    at Module.load (node:internal/modules/cjs/loader:1249:32)
    at Module._load (node:internal/modules/cjs/loader:1065:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:158:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\Users\\Arthur Araujo\\Desktop\\Solana-Memecoin-CLI\\src\\Token\\create.js'
  ]
}

Node.js v22.2.0

**Reply by outsmartchad** [reply]
Thanks you! Seem like you didn't npm install all the packages, 1. go back the first dir which containing the package.json, 2. npm install

**Reply by outsmartchad** [reply]
yes, we will open a discord channel, or you can directly send emails to me (chiwang1712@gmail.com)!

**Reply by outsmartchad** [reply]
<img width="1044" alt="image" src="https://github.com/ManofDiligence/Solana-Memecoin-CLI/assets/88470511/085fd559-9eca-4014-bae2-d654b14f77d3">
npm install all the dependencies our project needed

**Reply by pandaamx** [reply]
I've completed all the initial steps, but it's not working :(

```
cd Solana-Memecoin-CLI
nvm install v22.2.0
nvm use v22.2.0
npm install
```

my console:

```
PS C:\Users\Arthur Araujo\Desktop\Solana-Memecoin-CLI> npm i

up to date, audited 351 packages in 3s

63 packages are looking for funding
  run `npm fund` for details

4 moderate severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
```

**Reply by pandaamx** [reply]
![image](https://github.com/ManofDiligence/Solana-Memecoin-CLI/assets/171448628/24dad3f1-46e4-45e2-b40d-57bc29c896fa)

**Reply by pandaamx** [reply]
I tried, but another error occurred. It seems that there is a conflict between dep.:

```
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: memecoin-solana-cli@1.0.0
npm error Found: @metaplex-foundation/umi@0.8.10
npm error node_modules/@metaplex-foundation/umi
npm error   peer @metaplex-foundation/umi@">= 0.8.2 < 1" from @metaplex-foundation/mpl-candy-machine@6.0.1
npm error   node_modules/@metaplex-foundation/mpl-candy-machine
npm error     @metaplex-foundation/mpl-candy-machine@"^6.0.1" from the root project
npm error   peer @metaplex-foundation/umi@"^0.8.2" from @metaplex-foundation/mpl-token-metadata@3.0.0-alpha.27
npm error   node_modules/@metaplex-foundation/mpl-token-metadata
npm error     @metaplex-foundation/mpl-token-metadata@"3.0.0-alpha.27" from @metaplex-foundation/mpl-candy-machine@6.0.1
npm error     node_modules/@metaplex-foundation/mpl-candy-machine
npm error       @metaplex-foundation/mpl-candy-machine@"^6.0.1" from the root project
npm error   1 more (@metaplex-foundation/mpl-toolbox)
npm error
npm error Could not resolve dependency:
npm error @metaplex-foundation/umi-bundle-defaults@"*" from the root project
npm error
npm error Conflicting peer dependency: @metaplex-foundation/umi@0.9.1
npm error node_modules/@metaplex-foundation/umi
npm error   peer @metaplex-foundation/umi@"^0.9.1" from @metaplex-foundation/umi-bundle-defaults@0.9.1
npm error   node_modules/@metaplex-foundation/umi-bundle-defaults
npm error     @metaplex-foundation/umi-bundle-defaults@"*" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error C:\Users\Arthur Araujo\AppData\Local\npm-cache\_logs\2024-06-02T04_04_36_913Z-eresolve-report.txt

npm error A complete log of this run can be found in: C:\Users\Arthur Araujo\AppData\Local\npm-cache\_logs\2024-06-02T04_04_36_913Z-debug-0.log
```

**Reply by outsmartchad** [reply]
we're finding way to fix it 🤔, maybe i should unify all the dependencies of the project, stay tuned for the updates!

**Reply by pandaamx** [reply]
I'm here if u need help!

**Reply by rektcapital** [reply]
<img width="1348" alt="image" src="https://github.com/ManofDiligence/Solana-Memecoin-CLI/assets/107988448/0d929d6d-90c1-4b8c-a816-a41822e7c6b9">
@pandaamx @ManofDiligence 
1. rm -rf node_modules
2. rm package-lock.json
3. npm cache clean --force  
4. npm install 
5. change "@metaplex-foundation/umi" to "^0.9.1" in package.json
6. npm install @metaplex-foundation/umi-bundle-defaults --force
7. you notice that "@metaplex-foundation/umi-bundle-defaults": "^0.9.1", added into package.json
8. run it now (node src/Token/create -h) 
<img width="1448" alt="image" src="https://github.com/ManofDiligence/Solana-Memecoin-CLI/assets/107988448/63b6afd2-8ca5-420b-be5d-f2f98cdb5313">

**Reply by rektcapital** [reply]
i updated the packages.

**Reply by pandaamx** [reply]
```
Error: Cannot find module '@solana-developers/helpers'
Require stack:
- C:\Users\Arthur Araujo\Desktop\Solana-Memecoin-CLI\src\Token\create.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1186:15)
    at Module._load (node:internal/modules/cjs/loader:1012:27)
    at Module.require (node:internal/modules/cjs/loader:1271:19)
    at require (node:internal/modules/helpers:123:16)
    at Object.<anonymous> (C:\Users\Arthur Araujo\Desktop\Solana-Memecoin-CLI\src\Token\create.js:24:39)
    at Module._compile (node:internal/modules/cjs/loader:1434:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1518:10)
    at Module.load (node:internal/modules/cjs/loader:1249:32)
    at Module._load (node:internal/modules/cjs/loader:1065:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:158:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\Users\\Arthur Araujo\\Desktop\\Solana-Memecoin-CLI\\src\\Token\\create.js'
  ]
}
```

and after installing:

```
C:\Users\Arthur Araujo\Desktop\Solana-Memecoin-CLI\node_modules\base-x\src\index.js:69
    if (typeof source !== 'string') { throw new TypeError('Expected String') }
                                      ^

TypeError: Expected String
    at decodeUnsafe (C:\Users\Arthur Araujo\Desktop\Solana-Memecoin-CLI\node_modules\base-x\src\index.js:69:45)
    at Object.decode (C:\Users\Arthur Araujo\Desktop\Solana-Memecoin-CLI\node_modules\base-x\src\index.js:111:18)
    at Object.<anonymous> (C:\Users\Arthur Araujo\Desktop\Solana-Memecoin-CLI\src\helpers\config.js:24:43)
    at Module._compile (node:internal/modules/cjs/loader:1434:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1518:10)
    at Module.load (node:internal/modules/cjs/loader:1249:32)
    at Module._load (node:internal/modules/cjs/loader:1065:12)
    at Module.require (node:internal/modules/cjs/loader:1271:19)
    at require (node:internal/modules/helpers:123:16)
    at Object.<anonymous> (C:\Users\Arthur Araujo\Desktop\Solana-Memecoin-CLI\src\Token\create.js:33:5)

Node.js v22.2.0
```

Is there something wrong with my config. files now?

**Reply by pandaamx** [reply]
> i updated the packages.

pulling and trying again,,

**Reply by outsmartchad** [reply]
you can delete the line of @solana-developers/helpers, yes, there is something wrong with ur config.js, you're close to run the code!

**Reply by pandaamx** [reply]
Yeahhh, it's working! 🚀
I created my first token
Thanks a lot!

**Reply by outsmartchad** [reply]
nice, play with other functions too!

**Reply by outsmartchad** [reply]
some functions like buy, sell is slow sometimes, we're working on optimize these by adding priority fee to transactions, and tips to solana validator, stay tuned! The priority fee options is not work right now.

**Reply by pandaamx** [reply]
hmmm.. So do fees help optimize these transactions? I did not know. However, for my initial project I wanted to provide the token without fees, like PEPE ("No Taxes, No Bullshit. It’s that simple.")

**Reply by pandaamx** [reply]
Another question. Does the purchasing system allow me to use https://raydium.io/swap/ to buy, for example?

**Reply by outsmartchad** [reply]
> hmmm.. So do fees help optimize these transactions? I did not know. However, for my initial project I wanted to provide the token without fees, like PEPE ("No Taxes, No Bullshit. It’s that simple.")

the priority fee helps you to land transaction faster on Solana, yes, this is SPL (Solana Program Library) following fungible token standard on Solana, anyone can trade or transfer to anyone without any tax.

**Reply by outsmartchad** [reply]
> Another question. Does the purchasing system allow me to use https://raydium.io/swap/ to buy, for example?

yes, the swap function is based on raydium swap api, and it's free, and no limit

**Reply by outsmartchad** [reply]
> hmmm.. So do fees help optimize these transactions? I did not know. However, for my initial project I wanted to provide the token without fees, like PEPE ("No Taxes, No Bullshit. It’s that simple.")

we are creating the function to create a market for ur token, in order to let anyone to trade it.

**Reply by outsmartchad** [reply]
Feel free to contribute the project, you can also add your own functions which might be useful for any developers on Solana

**Reply by pandaamx** [reply]
For sure! I noticed that you have other projects in development, which is why I really wanted to join Discord to try to contribute in some way and help you out. I'm also a dev and I really want to delve deeper into this area

**Reply by outsmartchad** [reply]
> For sure! I noticed that you have other projects in development, which is why I really wanted to join Discord to try to contribute in some way and help you out. I'm also a dev and I really want to delve deeper into this area

Thanks u for being here and helping the development this early! Yes, the discord server is initialising>>>>>， stay tuned!

**Reply by outsmartchad** [reply]
> For sure! I noticed that you have other projects in development, which is why I really wanted to join Discord to try to contribute in some way and help you out. I'm also a dev and I really want to delve deeper into this area

Yeah, I do think that dapp dev on Solana is the great thing to learn in 2024 and cosmos sdk is also a great skill to have if you want to be blockchain dev

**Reply by outsmartchad** [reply]
@rektcapital @pandaamx the dc channel we want! [https://discord.gg/bB2n9bJU](https://github.com/ManofDiligence/Solana-Memecoin-CLI/discussions/url)
